src/asl.ts

```diff
@@ -3020,72 +3235,69 @@ export class ASL {
-        // we assume that an array literal or a call would return a variable.
-        if (
-          ASLGraph.isLiteralValue(listOutput) &&
-          !Array.isArray(listOutput.value)
-        ) {
-          throw new SynthError(
-            ErrorCodes.Unexpected_Error,
-            "Expected input to map to be a variable reference or array"
-          );
-        }
+    return this.evalExprToJsonPath(expr.expr.expr, (listOutput) => {
+      // we assume that an array literal or a call would return a variable.
+      if (
+        ASLGraph.isLiteralValue(listOutput) &&
+        !Array.isArray(listOutput.value)
+      ) {
+        throw new SynthError(
+          ErrorCodes.Unexpected_Error,
+          "Expected input to map to be a variable reference or array"
+        );
+      }
 
-        return this.iterateArrayMethod(
-          callbackfn,
-          normalizeOutputToJsonPath().jsonPath,
-          true,
-          {
-            arrStr: "[null",
-          },
-          ({
-            tailJsonPath,
-            tailTarget,
-            iterationResult,
-            workingSpaceJsonPath,
-          }) => {
-            return {
-              Type: "Pass",
-              Parameters: {
-                [`${tailTarget}.$`]: `${tailJsonPath}`,
-                // const arrStr = `${arrStr},${JSON.stringify(arr[0])}`;
-                "arrStr.$": `States.Format('{},{}', ${workingSpaceJsonPath}.arrStr, States.JsonToString(${iterationResult}))`,
-              },
-              ResultPath: workingSpaceJsonPath,
-              Next: ASLGraph.DeferNext,
-            };
-          },
-          (workingSpaceJsonPath) => {
-            return {
-              startState: "format",
-              states: {
-                format: {
-                  Type: "Pass",
-                  // filterResult = { result: JSON.parse(filterResult.arrStr + "]") }
-                  Parameters: {
-                    // cannot use intrinsic functions in InputPath or Result
-                    "result.$": `States.StringToJson(States.Format('{}]', ${workingSpaceJsonPath}.arrStr))`,
-                  },
-                  ResultPath: workingSpaceJsonPath,
-                  Next: "set",
-                },
-                // filterResult = filterResult.result
-                set: {
-                  Type: "Pass",
-                  // the original array is initialized with a leading null to simplify the adding of new values to the array string "[null"+",{new item}"+"]"
-                  InputPath: `${workingSpaceJsonPath}.result[1:]`,
-                  ResultPath: workingSpaceJsonPath,
-                  Next: ASLGraph.DeferNext,
+      return this.iterateArrayMethod(
+        callbackfn,
+        listOutput.jsonPath,
+        true,
+        {
+          arrStr: "[null",
+        },
+        ({
+          tailJsonPath,
+          tailTarget,
+          iterationResult,
+          workingSpaceJsonPath,
+        }) => {
```

Comment:
I wonder if this would be easier to read if instead of passing many arguments, we passed a single object with properties for each of the arguments? That way meaning is given to them?

Reply:
Wow, a little formatting change wrapped a ton of unchanged code into this PR... the only part that changed here was the parameters in the arrow function and line 3252.

I also considered using an object instead. I liked that the arrow functions were in the order they were executed. I guess we could change the first 3-4 parameters to an object as they are not in order.

Reply:
No big deal, i'm sure it reads better when you spend more time in the code base. Didn't realize this code was unchanged.